### PR TITLE
docs(core): add guide for workspace schematics

### PR DESCRIPTION
**Pull request by brandonroberts**
_[Please make sure you have read the submission guidelines before posting an PR](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#submit-pr)_

> _Please make sure that your commit message follows our format._

> Example: `fix(nx): must begin with lowercase`

### CODE REVIEW

File: docs/shared/tools-workspace-schematics.md [hunk 1]
```diff
@@ -1 +1,284 @@
 # Using Workspace Schematics
+
+Workspace schematics provide a way to automate many tasks you regularly perform as part of your development workflow. Whether its scaffolding out components, features, or ensuring libraries are generated and structured in a certain way, schematics help you standardize these tasks in a consistent, and predictable way. Nx provides tooling around creating, and running custom schematics from within your workspace. This guide shows you how to create, run, and customize workspace schematics within your Nx workspace.
```
Comment (juristr) [inline docs/shared/tools-workspace-schematics.md]: Type: _"Whether **it is** scaffolding out components,..."_

I would maybe split this sentence as follows:

Maybe avoid repetition of "way" ?
_"Whether ....structured in a certain way, schematics help you standardize these tasks in a consistent, and predictable manner."_

File: docs/shared/tools-workspace-schematics.md [hunk 2]
```diff
@@ -1 +1,284 @@
...
+export default function(schema: any): Rule {
+  return chain([
+    externalSchematic('@nrwl/workspace', 'lib', {
+      name: schema.name
+    })
+  ]);
+}
+```
+
+The `chain` function takes a an array of rules and combines them into a single rule. You use this function to perform multiple operations against your workspace in a single schematic. The `externalSchematic` function allows you to call schematics provided from an installed npm package external.
```
Comment (juristr) [inline docs/shared/tools-workspace-schematics.md]: Is that "external" needed at the end of the sentence? I'd rather just write
_"...to call schematics provided **by an** installed npm package ~~external~~"._

File: docs/shared/tools-workspace-schematics.md [hunk 3]
```diff
@@ -1 +1,284 @@
...
+
+To run a schematic, invoke the `nx workspace-schematic` command with the name of the schematic.
+
+```sh
+nx workspace-schematic my-schematic mylib
+```
+
+## Creating custom rules
+
+Schematics provide an API for managing files within your workspace. You can use schematics to do such things as create, update, move, and delete files. Files with static or dynamic content can also be created.
```
Comment (juristr) [inline docs/shared/tools-workspace-schematics.md]: _"You can use schematics to **do things such as** create, update, move..."_

**Review by juristr — COMMENTED** [review]
I like it, this article was really needed. Here just some minor comments while reading through it.

Maybe I could contribute a video lesson to be published on the Nrwl Youtube channel that walks through these changes?

**Comment by brandonroberts** [comment]
@juristr thanks for the feedback! I addressed your review comments. A video lesson would be great! cc: @isaacplmann

**Comment by isaacplmann** [comment]
I have full confidence in @juristr 's video making skills.  Go for it.

**Comment by juristr** [comment]
Awesome 👍

**Review by FrozenPandaz — APPROVED** [review]
LGTM!

Let's merge this as is and iterate on it afterwards?

**Pull request merged**